AdobeStockAsset/Model/AssetRepository.php

```diff
@@ -96,16 +122,46 @@ public function __construct(
         $this->collectionProcessor = $collectionProcessor;
         $this->searchResultFactory = $searchResultFactory;
         $this->assetSaveService = $commandSave;
-        $this->assetKeywordRepository = $assetKeywordRepository;
+        $this->getAssetKeywords = $getAssetKeywords;
+        $this->saveAssetKeywords = $saveAssetKeywords;
+        $this->saveMediaAsset = $saveMediaAsset;
+        $this->mediaAssetInterfaceFactory = $mediaAssetInterfaceFactory;
     }
 
     /**
-     * @inheritdoc
+     * Save asset
+     *
+     * @param AssetInterface $asset
+     * @return void
+     * @throws \Magento\Framework\Exception\CouldNotSaveException
      */
     public function save(AssetInterface $asset): void
     {
+        $mediaGalleryAssetId = $this->saveMediaGalleryAsset($asset);
+        $asset->setMediaGalleryId($mediaGalleryAssetId);
         $this->assetSaveService->execute($asset);
-        $this->assetKeywordRepository->saveAssetKeywords($asset);
+        $this->saveAssetKeywords->execute([$asset]);
+    }
+
+    /**
+     * Save media gallery asset based on the adobe asset data.
+     *
+     * @param AssetInterface $asset
+     *
+     * @return int
+     * @throws \Magento\Framework\Exception\CouldNotSaveException
+     */
+    private function saveMediaGalleryAsset(AssetInterface $asset): int
+    {
+        /** @var MediaAssetInterface $mediaAsset */
+        $mediaAsset = $this->mediaAssetInterfaceFactory->create();
```

Comment:
You can use a factory to create `MediaAssetInterface` object instead of using setter methods.